File: services/log-processor/internal/model/log_event.go
package model

type LogEvent struct {
	EventID       string         `json:"eventId"`
	Timestamp     string         `json:"timestamp"`
	ReceivedAt    string         `json:"receivedAt"`
	WorkspaceID   string         `json:"workspaceId"`
	ProjectID     string         `json:"projectId"`
	ApplicationID string         `json:"applicationId"`
	Environment   string         `json:"environment"`
	Service       string         `json:"service"`
	Level         string         `json:"level"`
	Message       string         `json:"message"`
	TraceID       string         `json:"traceId,omitempty"`
	CorrelationID string         `json:"correlationId,omitempty"`
	Metadata      map[string]any `json:"metadata,omitempty"`
}

File: services/log-processor/TraceFlow.LogProcessor/Contracts/LogEvent.cs
namespace TraceFlow.LogProcessor.Contracts;

public sealed record LogEvent(
    Ulid EventId,
    Ulid WorkspaceId,
    Ulid ProjectId,
    Ulid ApplicationId,
    string Environment,
    string Service,
    int Level,
    string Message,
    DateTimeOffset Timestamp,
    string? TraceId,
    string? CorrelationId,
    Dictionary<string, object?>? Metadata);